### PR TITLE
CC: runtime-rs support pull image in guest

### PR DESCRIPTION
**Pull request by YushuoEdge**
Support cc in runtime-rs. Mainly including following works:

- Support pull image in guest.
- Handle empty rootfs. 

Fixes: #6346

Signed-off-by: Yushuo <y-shuo@linux.alibaba.com>

### CODE REVIEW

File: src/runtime-rs/crates/resource/src/rootfs/mod.rs [hunk 1]
```diff
@@ -59,60 +59,21 @@ impl RootFsResource {
...
-        root: &oci::Root,
+        root_path: &str,
         bundle_path: &str,
         rootfs_mounts: &[Mount],
-    ) -> Result<Arc<dyn Rootfs>> {
+        is_image_offload: bool,
+    ) -> Result<Option<Arc<dyn Rootfs>>> {
         match rootfs_mounts {
-            // if rootfs_mounts is empty
             mounts_vec if mounts_vec.is_empty() => {
```
Comment (Tim-0731-Hzt) [inline src/runtime-rs/crates/resource/src/rootfs/mod.rs:68]: I think this change may cause it doesn't work for `virt-container`, maybe consider another way to make it work for both.

Comment (YushuoEdge) [reply src/runtime-rs/crates/resource/src/rootfs/mod.rs:68]: For which case, `virt_container` will not work?

File: src/runtime-rs/crates/service/src/image_service.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    handler: Arc<RuntimeHandlerManager>,
+}
+
+impl ImageService {
+    pub(crate) fn new(handler: Arc<RuntimeHandlerManager>) -> Self {
+        Self { handler }
+    }
+}
+
+async fn handler_message<TtrpcReq, TtrpcResp>(
```
[the remark this answers is not in the document]
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/service/src/image_service.rs:30]: I agree, but this function is the same as the one in `task_service.rs`, I think they could share the same code.

**Comment by studychao** [comment]
please fix the confilicting files.

**Comment by YushuoEdge** [comment]
The static check problem [make test under runk] will also appear on the main branch when I test on my personal machine.  It seems related to seccomp. My environment:

```
OS:
$ cat /proc/version
Linux version 5.10.156 (...) (gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #4 SMP Thu Dec 1 15:07:39 CST 2022

Rust:
$ rustc --version
rustc 1.66.0 (69f9c33d7 2022-12-12)

seccomp:
$ apt-get install seccomp
Reading package lists... Done
Building dependency tree       
Reading state information... Done
seccomp is already the newest version (2.5.1-1ubuntu1~20.04.2).
```

Has anyone experienced this?

**Comment by studychao** [comment]
Is that possible to raise a PR for merging to Kata main branch instead of CCv0 branch?

**Comment by YushuoEdge** [comment]
> Is that possible to raise a PR for merging to Kata main branch instead of CCv0 branch?

OK. Then should this PR be closed?

**Comment by studychao** [comment]
> > Is that possible to raise a PR for merging to Kata main branch instead of CCv0 branch?
> 
> OK. Then should this PR be closed?

Yes please.

**Pull request closed without merging**